src/bson.rs

```diff
@@ -838,3 +845,19 @@ impl From<DateTime<Utc>> for UtcDateTime {
         UtcDateTime(x)
     }
 }
+
+/// Represents a BSON regular expression.
+#[derive(Debug, Clone, PartialEq)]
+pub struct RegExp {
+    /// The regex pattern to match.
+    pub pattern: String,
+
+    /// The options for the regex.
+    ///
+    /// Options are identified by characters, which must be stored in
+    /// alphabetical order. Valid options are 'i' for case insensitive matching, 'm' for
+    /// multiline matching, 'x' for verbose mode, 'l' to make \w, \W, etc. locale dependent,
+    /// 's' for dotall mode ('.' matches everything), and 'u' to make \w, \W, etc. match
+    /// unicode.
+    pub options: String,
```

Comment:
would we ever want to consider introducing an options type here to enforce the alphabeticalness or so users wouldn't have to remember the various flags?

Reply:
I prefer leaving it like this; I don't think the BSON regex type is going to get used enough to warrant the extra work. If anything, I'm leaning towards just linking to the MongoDB documentation for details about the options rather than putting them here; I honestly don't know if they've changed at all since this comment was first written.